test/conformance/kernel/urKernelGetInfo.cpp

```diff
@@ -3,70 +3,128 @@
-        } else {
-            ASSERT_TRUE(returned_attributes.empty());
-        }
-        break;
-    }
-    default:
-        break;
+}
+
+TEST_P(urKernelGetInfoTest, ReferenceCount) {
+    auto property_name = UR_KERNEL_INFO_REFERENCE_COUNT;
+    size_t property_size = 0;
+
+    ASSERT_SUCCESS(
+        urKernelGetInfo(kernel, property_name, 0, nullptr, &property_size));
+    ASSERT_EQ(property_size, sizeof(uint32_t));
+
+    std::vector<char> property_value(property_size);
+    ASSERT_SUCCESS(urKernelGetInfo(kernel, property_name, property_size,
+                                   property_value.data(), nullptr));
+
+    auto returned_reference_count =
+        reinterpret_cast<uint32_t *>(property_value.data());
+    ASSERT_GT(*returned_reference_count, 0U);
+}
+
+TEST_P(urKernelGetInfoTest, Context) {
+    auto property_name = UR_KERNEL_INFO_CONTEXT;
+    size_t property_size = 0;
+
+    ASSERT_SUCCESS(
+        urKernelGetInfo(kernel, property_name, 0, nullptr, &property_size));
+    ASSERT_EQ(property_size, sizeof(ur_context_handle_t));
+
+    std::vector<char> property_value(property_size);
+    ASSERT_SUCCESS(urKernelGetInfo(kernel, property_name, property_size,
+                                   property_value.data(), nullptr));
+
+    auto returned_context =
+        reinterpret_cast<ur_context_handle_t *>(property_value.data());
+    ASSERT_EQ(context, *returned_context);
+}
+
+TEST_P(urKernelGetInfoTest, Program) {
+    auto property_name = UR_KERNEL_INFO_PROGRAM;
+    size_t property_size = 0;
+
+    ASSERT_SUCCESS(
+        urKernelGetInfo(kernel, property_name, 0, nullptr, &property_size));
+    ASSERT_EQ(property_size, sizeof(ur_program_handle_t));
+
+    std::vector<char> property_value(property_size);
+    ASSERT_SUCCESS(urKernelGetInfo(kernel, property_name, property_size,
+                                   property_value.data(), nullptr));
+
+    auto returned_program =
+        reinterpret_cast<ur_program_handle_t *>(property_value.data());
+    ASSERT_EQ(program, *returned_program);
+}
+
+TEST_P(urKernelGetInfoTest, Attributes) {
+    auto property_name = UR_KERNEL_INFO_ATTRIBUTES;
+    size_t property_size = 0;
+
+    ASSERT_SUCCESS(
+        urKernelGetInfo(kernel, property_name, 0, nullptr, &property_size));
+
+    std::vector<char> property_value(property_size);
+    ASSERT_SUCCESS(urKernelGetInfo(kernel, property_name, property_size,
+                                   property_value.data(), nullptr));
+
+    auto returned_attributes = std::string(property_value.data());
+    ur_platform_backend_t backend;
+    ASSERT_SUCCESS(urPlatformGetInfo(platform, UR_PLATFORM_INFO_BACKEND,
+                                     sizeof(backend), &backend, nullptr));
+    if (backend == UR_PLATFORM_BACKEND_OPENCL ||
+        backend == UR_PLATFORM_BACKEND_LEVEL_ZERO) {
+        // Older intel drivers don't attach any default attributes and newer ones force walk order to X/Y/Z using special attribute.
```

Comment:
never noticed that clang-format doesn't touch comments like this, would prefer if it was manually formatted to maintain the < 80 per line
```suggestion
        // Older intel drivers don't attach any default attributes and newer
        // ones force walk order to X/Y/Z using special attribute.
```